src/pages/events.tsx

```diff
@@ -0,0 +1,176 @@
+import { EventDashboard } from "@/components/EventsDashboard/EventDashboard";
+import { fetchBackend } from "@/lib/db";
+import { BiztechEvent } from "@/types/types";
+import { getCurrentUser } from "@aws-amplify/auth";
+import { ListIcon, SearchIcon, Bookmark } from "lucide-react";
+import React, { ChangeEvent, useEffect, useMemo, useRef, useState } from "react";
+
+interface registeredEvent {
+  "eventID;year": string;
+  [key: string | symbol]: unknown;
+}
+
+interface EventProps {
+  events: BiztechEvent[];
+}
+
+const filterStates = {
+  all: "all",
+  saved: "saved",
+};
+
+export default function Page({ events }: EventProps) {
+  const [searchField, setSearchField] = useState("");
+  const [filterState, setFilterState] = useState("all");
+  const isCooldownRef = useRef(false);
+  // these useStates will be empty arrays by default, but currently have mocks before i verify backend integration works
+  const [saved, setSaved] = useState<string[]>([]);
+  const [registered, setRegistered] = useState<string[]>([]);
+  const [email, setEmail] = useState<string>("");
+
+  useEffect(() => {
+    console.log(events);
+    fetchData();
+  }, []);
+
+  const uiStateFilter = () => {
+    let filteredEvents: BiztechEvent[] = events;
+    if (filterState === filterStates.saved && email) {
+      filteredEvents = filteredEvents.filter((ev) => {
+        return saved.includes(`${ev.id};${ev.year}`);
+      });
+    }
+
+    filteredEvents = filteredEvents.filter((ev) => {
+      return ev.ename.startsWith(searchField);
+    });
+
+    return filteredEvents;
+  };
+
+  const displayedEvents = useMemo(() => uiStateFilter(), [uiStateFilter, filterState, searchField, saved]);
+
+  const fetchData = async () => {
+    try {
+      const { signInDetails } = await getCurrentUser();
+      const email = signInDetails?.loginId;
+      const user = await fetchBackend({ endpoint: `/users/${email}`, method: "GET" });
+      const registeredEvents = await fetchBackend({ endpoint: `/registrations?email=${email}`, method: "GET" });
+      setEmail(email ? email : "");
+      setSaved(user["favedEventsID;year"] ? user["favedEventsID;year"] : []);
+      setRegistered(
+        registeredEvents.data.map((event: registeredEvent) => {
+          return event["eventID;year"];
+        })
+      );
+    } catch (error) {
+      console.error(error);
+    }
+  };
```

Comment:
As Allan mentioned, we have a use-case on the old `bt-web` where people who have not yet registered for an account can browse the Events page as a Guest